src/controllers/todo-controller.ts

```diff
@@ -27,28 +25,70 @@ export class TodoController extends BaseController {
     this.router.post(
       `${this.basePath}`,
       createTodoValidator(this.appContext),
-      this.createTodo,
+      this.createTodo
+    );
+
+    this.router.delete(
+      `${this.basePath}/:id`,
+      deleteTodoValidator(this.appContext),
+      this.deleteTodo
     );
   }
 
-  private createTodo = async (req:ExtendedRequest, res: Response, next: NextFunction) => {
+  private createTodo = async (
+    req: ExtendedRequest,
+    res: Response,
+    next: NextFunction
+  ) => {
     const failures: ValidationFailure[] = Validation.extractValidationErrors(
-      req,
+      req
     );
     if (failures.length > 0) {
       const valError = new Errors.ValidationError(
-        res.__('DEFAULT_ERRORS.VALIDATION_FAILED'),
-        failures,
+        res.__("DEFAULT_ERRORS.VALIDATION_FAILED"),
+        failures
       );
       return next(valError);
     }
 
     const { title } = req.body;
     const todo = await this.appContext.todoRepository.save(
       new Todo({
-       title
-      }),
+        title,
+      })
     );
     res.status(201).json(todo.serialize());
-  }
+  };
+
+  private deleteTodo = async (
+    req: ExtendedRequest,
+    res: Response,
+    next: NextFunction
+  ) => {
+    const failures: ValidationFailure[] = Validation.extractValidationErrors(
+      req
+    );
+    if (failures.length > 0) {
+      const valError = new Errors.ValidationError(
+        res.__("DEFAULT_ERRORS.VALIDATION_FAILED"),
+        failures
+      );
+      return next(valError);
+    }
+
+    const { id } = req.params;
+    const todo = await this.appContext.todoRepository.count({
```

Comment:
Controller should not do this.

Also, we should directly issue a delete command to db and determine appropriate response. DB operations can tell you what happened rather than querying it explicitly at application layer.